pet-api/pet-api/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using pet_api.Domain.Entities.Pagination;
+using pet_api.Domain.Entities;
+using pet_api.Domain.Interfaces;
+
+namespace pet_api.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class ProductsController : ControllerBase
+    {
+        private readonly IUnitOfWork _unitOfWork;
+
+        public ProductsController(IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+        }
+        [HttpGet("{id:int}")]
+        public async Task<ActionResult> GetProductById(int id)
+        {
+            Product product = await _unitOfWork.ProductRepository.GetById(id);
+            if (product == null)
+            {
+                return NotFound();
+            }
+            return Ok(product);
+        }
+
+        [HttpGet]
+        public async Task<ActionResult> GetProducts([FromQuery] ProductParameters productParameters)
```

Comment:
add type for ActionResult

Reply:
ActionResult contains IEnumerable<Product> parameter